### PR TITLE
Feat/subnets contract pt 2

### PR DESCRIPTION
**Pull request by pavitthrap**
### Description
Adds functions to example subnets contract: `deposit-ft`, `deposit-nft`, `withdraw-ft`, and `withdraw-nft`. 
Adds clarinet testing for new methods.

### Applicable issues
- fixes half of #13 

### Additional info (benefits, drawbacks, caveats)
- deposit-ft/-nft does not currently take in the hyperchains contract ID of the asset. Might need to do so in order to complete the issue, but I thought I could make such an alteration in the second PR that addresses this issue

### Checklist
- [x] Test coverage for new or modified code paths

### CODE REVIEW

File: core-contracts/contracts/helper/nft-trait-standard.clar [hunk 1]
```diff
@@ -0,0 +1,15 @@
+(define-trait nft-trait
```
Comment (gregorycoppola) [inline core-contracts/contracts/helper/nft-trait-standard.clar:1]: high-level question about this interface set-up

Who will implement an instance of this and when?

Like, is there one per each type of sub-token?

Can we just make a single, non-abstract interface that will work for all tokens, so users don't have that much to do?

**Review by gregorycoppola — COMMENTED** [review]
Hello..

Mostly wondering at a high level how the trait structure is supposed to work?

Who makes an implementation of e.g. `nft-trait`? Is it possible to handle all nft's using a concrete implementation?

File: core-contracts/contracts/helper/nft-trait-standard.clar (hunk 1 above)
Comment (pavitthrap) [reply core-contracts/contracts/helper/nft-trait-standard.clar:1]: > Hello..
> 
> Mostly wondering at a high level how the trait structure is supposed to work?
> 
> Who makes an implementation of e.g. `nft-trait`? Is it possible to handle all nft's using a concrete implementation?

The nft/ft trait is defined in a SIP: 
- https://github.com/stacksgov/sips/blob/main/sips/sip-009/sip-009-nft-standard.md
- https://github.com/stacksgov/sips/blob/main/sips/sip-010/sip-010-fungible-token-standard.md

File: core-contracts/tests/subnets/subnets_test.ts [hunk 2]
```diff
@@ -71,3 +71,248 @@ Clarinet.test({
...
+            Tx.contractCall("subnets", "deposit-nft-asset",
+                [
+                    types.uint(1),
+                    types.principal(charlie.address),
+                    types.principal(nft_contract.contract_id),
+                ],
+                charlie.address),
+        ]);
+        assertEquals(block.height, 3);
+        block.receipts[0].result
```
[the remark this answers is not in the document]
Comment (pavitthrap) [reply core-contracts/tests/subnets/subnets_test.ts:112]: Added that!

**Comment by gregorycoppola** [comment]
LGTM, aside from kantai's comments.

**Review by kantai — APPROVED** [review]

**Review by gregorycoppola — APPROVED** [review]
nice job thanks

**Pull request merged**